backend/donor-service/utils/utils.js

```diff
@@ -66,8 +67,46 @@ const convertToSocialShareResponse = (entityName, userData) => {
         }, {});
 }
 
+async function checkForPledgeStatusAndReturnError (abhaNumber) {
+    let pledgeStatus = await getPledgeStatus(abhaNumber);
+    switch(pledgeStatus) {
```

Comment:
what happens if the pledger is first time pledger i.e NOT_PLEDGED ?

Reply:
Have added the case.